### DISCUSSION TITLE
Has v1.1 betas 3 and 4 been better or worse?

### DISCUSSION THREAD

**Discussion by vincentneo** [vincentneo/LosslessSwitcher#19] [Polls]
Vote below, and comment your thoughts and observations on the new betas!

**Comment by vladimir-krstic** [comment]
Better it doesn't jump between sample rates like crazy, but still not good enough for me. 
Issues I've encountered:
- Interrupting gapless playback from time to time.
- Audio cuts off while it tries to match the sample rate.

**Reply by vincentneo** [reply]
just to clarify, by gapless playback, I am assuming it is the tracks of the same album of same sample rate?

**Reply by vladimir-krstic** [reply]
That's right, on the same album, same sample rate. 

By the way, I really appreciate you doing this! Thanks a ton! 🙏

**Comment by andresandiah** [comment]
For me it has been way better when using it with my FiiO Q3, sample rate change has been accurate and fast the 90% of times. Thanks for all your effort!.
Will try later with my UAD Apollo X4 which I mainly use for production purposes but ocasionally listen music with.

Thanks a ton!

**Comment by jeffry1984** [comment]
Much better mate.
It does sometimes get glitchy on my qutest when 192khz or 96khz songs are in the queue after a 44.1kHz song. But usually works 8/10.
Only thing it doesn't do on my end is if I choose any song in my library to "play next" overriding apple musics randomization, then it will only choose the bit rate of the previous song no matter what. Hope that makes sense.

But fantastic work mate. so glad you're doing this and so appreciative.

**Comment by ashafai** [comment]
How do I get the latest version? I currently have 1.0 installed. 

Thanks.

**Reply by vincentneo** [reply]
You can get it here: https://github.com/vincentneo/LosslessSwitcher/releases/tag/1.1.0-beta4

**Comment by ashafai** [comment]
Thanks, Vincent. I just installed, and the following is what I am observing.

- LS Beta4
- MacBook Pro 2019
- Chord Hugo TT2 DAC

- Switching happens in less than 10 seconds, in most cases, which is much faster than before. I was observing around 15-20 seconds before.

- TT2 is accurately being switched, with exceptions noted below.

- Playing a 24/96 track, on some tracks it bounces back and forth between 48 7 96, but settles on 48, whereas on other 24/96 in the same album are recognized accurately and kept at 96.

- Playing a 24/192 track, had the same above observations as the 24/96 tracks.

Thank you for all the work, and I appreciate all the improvements. 😃

**Comment by MasterRahool** [comment]
Find that when going to hi-res tracks theres the behavior noted above - it switches to 48 then a second later finally switches to 96

this happens even within the same album (one track it was a split second, next i tried was a full second lag going from 96, 48, then realizing it should be 96)

edit: same on macbook or with a dac, and right now its just flicking around so much its too annoying to use

another issue i find is that when playing non-lossless tracks, the switcher just stays on whatever it last used. ie, 96khz track then go to a standard MP3....it still says 96khz lossless in the bar. If it cant default to (44.1?) then I think it should show blank or symbol if theres no lossless info to retrieve, ya?

**Reply by vincentneo** [reply]
Sorry @MasterRahool, haven't been monitoring this discussion for a while...


I think the feedback for defaulting to 44.1kHz for non lossless tracks should be doable, thanks for the suggestion!

**Comment by tomekbuszewski** [comment]
Hi, on a following config:

- iMac 2019 (Intel) 
- Cayin iDAC-9
- Beta 4

I am seeing an improvement on picking sample rate (it was picking 48kHz mostly before). It picks proper value now. But still doesn't sync bit rate, picks 16 every time.

**Reply by vincentneo** [reply]
Yup bit rate is a API limitation thus far, haven't found how Audio MIDI switches it yet (definitely not a 'Apple-preferred' way of switching, even if possible)

**Reply by vincentneo** [reply]
@tomekbuszewski, please try out release v1.1 beta 8

**Comment by MasterRahool** [comment]
Update on beta 4 since its actually working for me (unlike 5). I changed my MIDI settings to be **24-bit** instead of the default 32-bit I had it set to and so far I havent had a single switching issue.

knock on wood, but been testing tracks n stuff for over an hour and havent been able to make it really mess up like previously

that said, i did notice going from 176 or 96k to 192 tracks, it would do the "switch to 192, then 48, then back to 192" thing, and I'm having some issues when my EQ app is running causing it to display the wrong sample rate regardless whats going on (pictured below), and not switch, but as I'm typing this it's working again...
<img width="949" alt="Screenshot 2022-09-14 at 3 11 38 PM" src="https://user-images.githubusercontent.com/104704170/190242530-f033d8ba-bf52-4bd7-b1c8-03bd9f882c62.png">


system: M1 macbook on OS Ventura

**Reply by vincentneo** [reply]
Anything above 96 seems to have more odd switching behaviour, have personally experienced before.
